_includes/macros/share.html

```diff
@@ -31,72 +31,101 @@
-            {% set share_twitter_url = share_twitter_url + "&amp;hashtags=" + options.hashtags %}
+            {% set _share_twitter_url = _share_twitter_url + "&amp;hashtags=" + options.hashtags %}
         {% endif %}
     {% endif %}
-    {{ share_twitter_url }}
+    {{ _share_twitter_url }}
 {% endmacro %}
 
 
 {# ==========================================================================
 
-   share()
+   share.render()
 
    ==========================================================================
 
    Description:
 
-   Make share icons when given:
+   Make share icons when given a dict of options (optional):
 
-   title:                      The title of the thing to share. Used by the
-                               'share via email'.
+       title:           The title of the thing to share. Used by the
+                        'share via email' and 'share via LinkedIn'
 
-   hide_heading (optional):    Toggles the heading. Defaults to false.
+       heading:         Heading text for the social-share module
 
-   twitter_options (optional): See `options` under the `share_twitter_url()`
-                               macro.
+       hide_heading:    Toggles the heading. Defaults to false.
 
-   share_options (optional):   Object used to modify share markup.
+       show_email:      Toggles the email share icon
+
+       show_facebook:   Toggles the Facebook share icon
+
+       show_twitter:    Toggles the Twitter share icon
+
+       show_linkedin:   Toggles the LinkedIn share icon
+
+       twitter_options: See `options` under the `share_twitter_url()`
+                        macro.
 
    ==========================================================================
 
    Example:
 
-       {{ share('Title for email sharing', false,
-                { 'text': 'I can write my custom tweet text here' }) }}
+       {{ share.render({
+            'title': 'Title for email sharing',
+            'heading': 'Share this page',
+            'hide_heading': false,
+            'show_linkedin': true,
+            'twitter_options': {
+                'text': 'I can write my custom tweet text here'
+            }
+        }) }}
 
    ========================================================================== #}
 
-{% macro share(title, hide_heading=false, twitter_options={}, share_options={}) %}
-    {% set text = share_options.text | default('Share this page') %}
-    {% set additional_classes = share_options.additional_classes  %}
-
-    <aside {{ ('class=' ~ additional_classes) if additional_classes }}>
-        <h1 class="h5 {{'u-visually-hidden' if hide_heading}}">{{ text }}</h1>
-        <ul class="post_share list__horizontal">
+{% macro render(options) %}
+    {% set settings = {
+        'heading': 'Share this post',
+        'hide_heading': false,
+        'show_email': true,
+        'show_facebook': true,
+        'show_twitter': true,
+        'show_linkedin': false
+    } %}
+  {% set _ignore = settings.update(options) %}
```

Comment:
Ahh okay. Looks like we could use an [expression statement](http://jinja.pocoo.org/docs/dev/templates/#expression-statement) _if_ we had the expression-statement extension loaded, which it doesn't look like we do.